crates/starknet/src/state/state_diff.rs

```diff
@@ -0,0 +1,238 @@
+use std::collections::HashMap;
+
+use starknet_in_rust::state::cached_state::CachedState;
+use starknet_in_rust::state::in_memory_state_reader::InMemoryStateReader;
+use starknet_in_rust::state::StateDiff as StarknetInRustStateDiff;
+use starknet_in_rust::utils::subtract_mappings;
+use starknet_in_rust::CasmContractClass;
+use starknet_types::contract_class::ContractClass;
+use starknet_types::felt::{ClassHash, Felt};
+
+use crate::error::Result;
+
+/// This struct is used to store the difference between state modifications
+#[derive(PartialEq, Default, Debug, Clone)]
+pub struct StateDiff {
+    // data taken from starknet_in_rust
+    pub(crate) inner: StarknetInRustStateDiff,
+    // class hash to compiled_class_hash difference, used when declaring contracts
+    // that are different from cairo 0
+    pub(crate) class_hash_to_compiled_class_hash: HashMap<ClassHash, ClassHash>,
+    // declare contracts that are not cairo 0
+    pub(crate) declared_contracts: HashMap<ClassHash, CasmContractClass>,
+    // cairo 0 declared contracts
+    pub(crate) cairo_0_declared_contracts: HashMap<ClassHash, ContractClass>,
+}
+
+impl Eq for StateDiff {
+    fn assert_receiver_is_total_eq(&self) {}
```

Comment:
to be able to derive a trait on a struct all of the fields should implement the target trait. StarknetInRustStateDiff doesn't derive Eq trait

Reply:
My first question was more like "Why do we need the method `assert_receiver_is_total_eq` if it's not used anywhere?"
The other question was more like "Why do we need to do `impl Eq` instead of `#[derive(Eq)]` like we did with other traits in L14?"

Reply:
because on this struct you cannot write #[derive(Eq)], because of the above reason

Reply:
Ok, now I get the answer to the second question.

Reply:
Ok, about the first question, I didn't realize this `assert_receiver_is_total_eq` is an actual pre-defined function of the `Eq` trait, I thought it was something you accidentally left here. But how come in other places in the code we can do things like `impl Eq for InvokeTransactionV1 {}` (so an empty `impl` body).

Reply:
I will remove the empty implementation

Reply:
I also don't know why this method is empty and what to assert in the implementation

Reply:
anyway [2a3e99e](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/103/commits/2a3e99e4611cac982448e295352b6cf7629fd60e)